src/core/vertexmodel.cpp

```diff
@@ -696,6 +727,25 @@ void VertexModel::setCurrentPoint( const QgsPoint &point )
 
   Vertex &vertex = mVertices[mCurrentIndex];
 
+  // Calculate angles between edges
+
+  int startPoint = mCurrentIndex - 2;
+  if ( startPoint < 0 )
+  {
+    startPoint = vertexCount() + startPoint;
+  }
+  int endPoint = mCurrentIndex + 2;
+  if ( endPoint >= vertexCount() )
+  {
+    endPoint = endPoint - vertexCount();
+  }
+
+  double angle = calculateAngle( mVertices[startPoint].point, mVertices[mCurrentIndex].point, mVertices[endPoint].point );
+
+  qDebug() << "--------- angle = " << angle;
+
+  setAngleFonud( angle > snapToCommonAngleDegrees() - 1 && angle < snapToCommonAngleDegrees() + 1 );
```

Comment:
We discussed this yesterday. This wouldn't really work as a/ we wouldn't get a perfect coordinate snapped to angle in any case, and b/ we'd allow for angle 89 and 91 :scream: :wink: